opendevin/agent.py

```diff
@@ -89,17 +82,7 @@ def history(self) -> List[str]:
         return self._history
 
     @abstractmethod
-    def add_event(self, event: Event) -> None:
-        """
-        Adds an event to the agent's history.
-
-        Parameters:
-        - event (Event): The event to add to the history.
-        """
-        pass
-
-    @abstractmethod
-    def step(self, cmd_mgr: CommandManager) -> Event:
```

Comment:
I get rid of `add_event` method, with the hope that we can put all these "added events" into `State`, which is passed to agent's `.step` method, with the hope that it simplifies the workflow and make the code easier for people to understand?